fla/layers/mom.py

```
@@ -0,0 +1,613 @@
+                # recurrent_state[e] = state_e
+                for batch in range(state_e.shape[0]):
+                    if recurrent_state[e] is None:
+                        recurrent_state[e] = state_e
+                    elif hidden_states[e, batch].any():
+                        recurrent_state[e][batch] = state_e[batch]
+            o_list = torch.stack(o_list, dim=0)
+            o = reconstruct(o_list, indices=indices, sorted_indices=sorted_indices, batch_size=q.shape[1], seq_len=seq_len, topk=self.topk, routing_weights=routing_weights, mask=mask)
+
+        if self.shared_mem:
+            shared_o = self.shared_o(shared_hidden_states, attention_mask, recurrent_state, use_cache, conv_state_q, conv_state_k, conv_state_v)
+            o += shared_o
```

Pylint (3.3.7)

[error] Undefined variable 'o'

(E0602)